File: lib/Karve/CustomControls/DataGrid/Clases/DataGridOrder/DataGridOrdenColumnas.vb
Public Class DataGridOrdenColumnas
    Inherits System.Collections.ArrayList
    '*--------------------------------------------------------------------------
    '*  ESTA COLECCIÓN PERMITE REPES, POR ESO LA CREO ARRAY LIST
    '*--------------------------------------------------------------------------

    Public Overloads Sub Add(ByVal Columnas As DataGridordenColumna)
        Try
            Columnas.Item = Me.Count
            MyBase.Add(Columnas)
        Catch
        End Try
    End Sub

    Public Function DameOrder() As String
        Dim _sOrder As String = ""
        Dim RL As New DataGridOrdenColumna
        Dim _sClausula As String = ""
        Dim _sCampo As String = ""
        Dim _sOperador As String = ""
        '*====================================================================
        For Each RL In Me.ToArray
            If RL.Item > 0 Then _sOperador = ", "
            If RL.Expresion <> "" Then
                _sClausula = RL.Expresion + " " + RL.CriterioString
            ElseIf RL.ExpresionBD <> "" Then
                _sClausula = "(" & RL.ExpresionBD & ") " & RL.CriterioString
            ElseIf RL.AliasCampo <> "" Then
                _sClausula = RL.AliasCampo + " " + RL.CriterioString
            Else
                _sCampo = ""
                _sClausula = ""
                '*------------------------------------------------------------
                If RL.AliasTabla <> "" Then _sCampo = RL.AliasTabla & "."
                _sCampo += RL.Campo
                _sClausula += _sCampo + " " + RL.CriterioString
            End If
            _sOrder = _sOrder & _sOperador & _sClausula
        Next
        If _sOrder <> "" Then _sOrder = " ORDER BY " & _sOrder
        Return _sOrder
    End Function

End Class

File: src/Controls/KarveControls/DataGrid/DataGridHelpers/DataGridOrder/DataGridOrdenColumnas.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
public class DataGridOrdenColumnas : System.Collections.ArrayList
{
	//*--------------------------------------------------------------------------
	//*  ESTA COLECCIÓN PERMITE REPES, POR ESO LA CREO ARRAY LIST
	//*--------------------------------------------------------------------------

	public void Add(DataGridOrdenColumna Columnas)
	{
		try {
			Columnas.Item = this.Count;
			base.Add(Columnas);
		} catch {
		}
	}

	public string DameOrder()
	{
		string _sOrder = "";
		string _sClausula = "";
		string _sCampo = "";
		string _sOperador = "";
		//*====================================================================
		foreach (DataGridOrdenColumna RL in this) {
			if (RL.Item > 0)
				_sOperador = ", ";
			if (!string.IsNullOrEmpty(RL.Expresion)) {
				_sClausula = RL.Expresion + " " + RL.CriterioString;
			} else if (!string.IsNullOrEmpty(RL.ExpresionBD)) {
				_sClausula = "(" + RL.ExpresionBD + ") " + RL.CriterioString;
			} else if (!string.IsNullOrEmpty(RL.AliasCampo)) {
				_sClausula = RL.AliasCampo + " " + RL.CriterioString;
			} else {
				_sCampo = "";
				_sClausula = "";
				//*------------------------------------------------------------
				if (!string.IsNullOrEmpty(RL.AliasTabla))
					_sCampo = RL.AliasTabla + ".";
				_sCampo += RL.Campo;
				_sClausula += _sCampo + " " + RL.CriterioString;
			}
			_sOrder = _sOrder + _sOperador + _sClausula;
		}
		if (!string.IsNullOrEmpty(_sOrder))
			_sOrder = " ORDER BY " + _sOrder;
		return _sOrder;
	}

}
